src/plugins/intel_gna/src/transformations/aszp_decomposition.cpp

```diff
@@ -0,0 +1,231 @@
+	uint64_t N = input_shape[0]; // batchsize
+	uint64_t H = input_shape[1]; // no input channels (input depth)
+	uint64_t W = input_shape[2]; // input height
+	uint64_t C = input_shape[3]; // input widith
+
+	return std::make_tuple(N,H,W,C);
+}
+
+
+
+std::shared_ptr<ov::op::v0::Concat> ConcatZeros(
+	uint32_t pad_begin,
+	uint32_t pad_end,
+	std::shared_ptr<ov::Node> padding_const,
+	std::shared_ptr<ov::Node> input_node) {
+
+	OutputVector concat_vector;
+	if (pad_begin > pad_end) {
+		concat_vector.push_back(padding_const->output(0));
+		concat_vector.push_back(input_node->output(0));
+	}else {
+		concat_vector.push_back(input_node->output(0));
+		concat_vector.push_back(padding_const->output(0));	
+	}
+	return std::make_shared<ngraph::op::Concat>(concat_vector,1);
+
+}
+
+
+
+
+Output<Node> DecomposeHeight(Output<Node> input, ov::CoordinateDiff pads_begin, ov::CoordinateDiff pads_end, ov::Shape conv_input_shape) {
+	uint32_t Hbegin, Hend, Wbegin, Wend, Hpad, Wpad;
+	uint64_t N, C, H, W;
+    std::tie(Hbegin, Hend, Wbegin, Wend, Hpad, Wpad) = ExtractPadding(pads_begin, pads_end);
+    std::tie(N,H,W,C) = GetInputDimensions(conv_input_shape);
+
+	if (Hpad != 0) {
+		auto new_reshape   = CreateReshape(input, 2, Shape{H,W*C});
+		auto new_transpose = CreateTranspose(new_reshape->output(0));
+
+		auto padding_const = CreateZeroConst(Shape{W*C,Hpad});
+		auto new_concat    = ConcatZeros(Hbegin,Hend,padding_const,new_transpose);
+		
+		auto new_untranspose = CreateTranspose(new_concat->output(0));	// Shape {W*C,H+Hpad}
+
+		if (0 == Wpad) return CreateReshape(new_untranspose->output(0), 4, Shape{N, H+Hpad, W ,C})->output(0);
+		else return (new_untranspose->output(0));
+	}else return input;
+
+
+}
+
+
+
+Output<Node> DecomposeWidth(Output<Node> input, ov::CoordinateDiff pads_begin, ov::CoordinateDiff pads_end, ov::Shape conv_input_shape) {
+	uint32_t Hbegin, Hend, Wbegin, Wend, Hpad, Wpad;
+	uint64_t N, H, W, C;
+    std::tie(Hbegin, Hend, Wbegin, Wend, Hpad, Wpad) = ExtractPadding(pads_begin, pads_end); //todo cleanup. maybe get only what we need?
+    std::tie(N,H,W,C) = GetInputDimensions(conv_input_shape);
+
+	if (Wpad != 0) {
+		auto new_reshape   = CreateReshape(input, 2, Shape{(H+Hpad)*W, C});
+		auto new_transpose = CreateTranspose(new_reshape->output(0));
+		auto new_reshape2  = CreateReshape(new_transpose->output(0),2, Shape{C*(H+Hpad), W});
+
+		auto padding_const = CreateZeroConst( Shape{C*(H+Hpad),Wpad} );
+		auto new_concat    = ConcatZeros(Wbegin,Wend,padding_const,new_reshape2);
+
+		auto new_unshape2    = CreateReshape(new_concat->output(0), 2, Shape{C, (H + Hpad) * (W + Wpad)});
+		auto new_untranspose = CreateTranspose(new_unshape2->output(0));
+		auto new_unshape     = CreateReshape(new_untranspose->output(0), 4, {N, H+Hpad, W+Wpad ,C});
+
+		return new_unshape->output(0);
+	}else return input;
+}
+
+
+
+void TrimmPadding(ov::CoordinateDiff& pads_begin, ov::CoordinateDiff& pads_end) {
+	if ( pads_begin[0] > pads_end[0]) 
```

Comment:
please use `{ }` statements after conditions

Reply:
please fix it in all places

Reply:
done